extension/controller/backgroundScript.js

```diff
@@ -306,6 +310,18 @@ const messageListener = function(message, sender) {
         case "setStorage":
           await browser.storage.local.set(message.payload)
           break;
+        case "translateAsBrowse":
+
+          /*
+           * we received a request to translate as browse. so this means
+           * we need to record both the tabid, the website and the pagelanguage
+           * so that when there's a navigation in this tab, site and samelanguage
+           * we should automatically start the translation
+           */
+          translateAsBrowseMap.set(message.tabId, {
```

Comment:
shouldn't we remove this tabid from the map when the tab is unloaded?

Reply:
done